functions/Get-DbaDbFile.ps1

```diff
@@ -59,20 +68,54 @@ function Get-DbaDbFile {
         [Alias("Databases")]
         [object[]]$Database,
         [object[]]$ExcludeDatabase,
+        [parameter(ValueFromPipeline)]
+        [Microsoft.SqlServer.Management.Smo.Database[]]$SmoDatabase,
         [object[]]$InputObject,
         [Alias('Silent')]
         [switch]$EnableException
     )
 
+    begin {
+
+        # Used to ensure parameter validation checks only run once.
+        $initialParameterCheck = $true;
+    }
+
     process {
 
-        foreach ($instance in $sqlInstance) {
+        # The check only needs to be run once, but cannot be done in begin{} blcok since it doens't know if we'll receive pipeline input
+        if ($initialParameterCheck) {
+
+            if ($SqlInstance -and $SmoDatabase) {
+
+                Stop-Function -Message "You cannot specify both SqlInstance and SmoDatabase." -Continue -EnableException $EnableException;
+            }
+
+            if ($SmoDatabase -and $Database) {
+
+                Write-Message -Level Verbose -Message "Parameter Database is not being used since SmoDatabase was provided as well, using that instead.";
```

Comment:
Both can be used with no issue. If you do `InputObject += Get-DbaDatabase...` then it wouldn't matter which one they pick.